src/TypeSpec.Extension/Emitter.Csharp/src/lib/model.ts

```diff
@@ -648,12 +626,34 @@ export function getInputType(
                     Description: getDoc(program, value) ?? "",
                     Type: inputType,
                     IsRequired: !value.optional,
-                    IsReadOnly: isReadOnly,
-                    IsDiscriminator: false
+                    IsReadOnly: isReadOnly
                 } as InputModelProperty;
+
+                if (name === model.DiscriminatorPropertyName) {
+                    inputProp.IsDiscriminator = true;
+                    discriminatorPropertyDefined = true;
+                }
                 outputProperties.push(inputProp);
             }
         });
+
+        if (model.DiscriminatorPropertyName && !discriminatorPropertyDefined) {
```

Comment:
This also changes one thing: previously, the discriminator property will always be the first one in the property list.
Now it is almost always the last.
Could we also improve a bit so that we could minimize the changes? This change actually applies to every model with a discriminator

Reply:
How about we do it in two passes
First pass: see if there is a property defined as the discriminator. If no, we push one property created by us into the list
Second pass: add every property into the list